src/components/Select/Select.tsx

```diff
@@ -490,24 +490,24 @@ export const Select: FC<SelectProps> = React.forwardRef(
     };
 
     const getPills = (): JSX.Element => {
-      const selected: SelectOption[] = options.filter(
-        (opt: SelectOption) => opt.selected
+      const selected: SelectOption[] = options?.filter(
```

Comment:
similarly, if you provide an empty fallback, you can safely assume selected is an array below
```suggestion
      const selected: SelectOption[] = (options || []).filter(
```